DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1

```diff
@@ -87,73 +105,86 @@ function Set-TargetResource
         $($LocalizedData.ApplyingDNSServerAddressesMessage)
         ) -join '')
 
-    #Get the current DNS Server Addresses based on the parameters given.
-    $PSBoundParameters.Remove('Address')
-    $PSBoundParameters.Remove('Validate')
-    $currentAddress = (Get-DnsClientServerAddress @PSBoundParameters `
-        -ErrorAction Stop).ServerAddresses
-
-    #Check if the Server addresses are the same as the desired addresses.
-    [Boolean] $addressDifferent = (@(Compare-Object `
-            -ReferenceObject $currentAddress `
-            -DifferenceObject $Address `
-            -SyncWindow 0).Length -gt 0)
-
-    if ($addressDifferent)
+    # Get the current DNS Server Addresses based on the parameters given.
+    # Interface Alias may return multiple when passed WildCard (*)
+    $interfaceAliases = Get-NetAdapter -Name $InterfaceAlias
+    foreach ($alias in $interfaceAliases.InterfaceAlias)
     {
-        # Set the DNS settings as well
-        $Splat = @{
-            InterfaceAlias = $InterfaceAlias
-            Address = $Address
-            Validate = $Validate
+        $currentAddress = (Get-DnsClientServerAddress `
+                -InterfaceAlias $alias `
+                -AddressFamily $AddressFamily `
+                -ErrorAction Stop).ServerAddresses
+
+        # Check if the Server addresses are the same as the desired addresses.
+        [Boolean] $addressDifferent = (@(Compare-Object `
```

Comment:
I think you mean to change Set-TargetResource like this. (remove if clause.)

```powershell
function Set-TargetResource
{
<#
.SYNOPSIS 
Set Node configuration to desired state.

.DESCRIPTION
Set a new Server Address in the current node
#>
    [CmdletBinding()]
    param
    (    
        # IP Address that has to be set    
        [Parameter(Mandatory = $true)]
        [ValidateNotNullOrEmpty()]
        [String[]]$Address,

        [Parameter(Mandatory = $true)]
        [ValidateNotNullOrEmpty()]
        [String]$InterfaceAlias,

        [Parameter(Mandatory = $true)]
        [ValidateSet('IPv4', 'IPv6')]
        [String]$AddressFamily,
        
        [Boolean]$Validate = $false
    )

    Write-Verbose -Message ( @("$($MyInvocation.MyCommand): "
        $($LocalizedData.ApplyingDNSServerAddressesMessage)
        ) -join '')

    # Get the current DNS Server Addresses based on the parameters given.
    # Interface Alias may return multiple when passed WildCard (*)
    $interfaceAliases = Get-NetAdapter -Name $InterfaceAlias
    foreach ($alias in $interfaceAliases.InterfaceAlias)
    {
        # Set the DNS settings as well
        $Splat = @{
            InterfaceAlias = $alias
            Address = $Address
            Validate = $Validate
        }
        Set-DnsClientServerAddress @Splat `
            -ErrorAction Stop
    }
}
```

Agree. But how we handle which interface don't need to set? Do you accept change both "already configured DNS" and "not configured DNS" when 1 of interface required to be sat?
If so, it should be better fail fast on Test-TargetResource too,
https://github.com/guitarrapc/xNetworking/blob/AddWildcardSupport/DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1#L211-L227

Something like this.

```powershell
function Test-TargetResource
{
<#
.SYNOPSIS 
Test Node configuration is in desired state or not.
.DESCRIPTION
Test if the given Server Address is among the current node's Server Address collection.
#>
    [OutputType([System.Boolean])]
    [CmdletBinding()]
    param
    (        
        [Parameter(Mandatory = $true)]
        [ValidateNotNullOrEmpty()]
        [String[]]$Address,

        [Parameter(Mandatory = $true)]
        [ValidateNotNullOrEmpty()]
        [String]$InterfaceAlias,

        [Parameter(Mandatory = $true)]
        [ValidateSet('IPv4', 'IPv6')]
        [String]$AddressFamily,
        
        [Boolean]$Validate = $false        
    )

    Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
        $($LocalizedData.CheckingDNSServerAddressesMessage)
        ) -join '' )

    # Validate the Settings passed
    Foreach ($ServerAddress in $Address) {       
        Test-ResourceProperty `
            -Address $ServerAddress `
            -AddressFamily $AddressFamily `
            -InterfaceAlias $InterfaceAlias
    }

    # Get the current DNS Server Addresses based on the parameters given.
    # Interface Alias may return multiple when passed WildCard (*)
    $interfaceAliases = Get-NetAdapter -Name $InterfaceAlias
    foreach ($alias in $interfaceAliases.Name)
    {
        $currentAddress = (Get-DnsClientServerAddress `
            -InterfaceAlias $alias `
            -AddressFamily $AddressFamily `
            -ErrorAction Stop).ServerAddresses

        # Check if the Server addresses are the same as the desired addresses.
        [Boolean] $addressDifferent = (@(Compare-Object `
                -ReferenceObject $currentAddress `
                -DifferenceObject $Address `
                -SyncWindow 0).Length -gt 0)

        if ($addressDifferent)
        {
            # fail fast return when one of interface Test result was false.
            $desiredConfigurationMatch.Add($false)
            Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
                $($LocalizedData.DNSServersNotCorrectMessage) `
                    -f ($Address -join ','),($currentAddress -join ',')
                ) -join '' )
            return $false;
        }
        else 
        {
            # Test will return true in this case
            Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
                $($LocalizedData.DNSServersSetCorrectlyMessage)
                ) -join '' )
        }
    }
}
```